theseus/embodied/robot/FK.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+
+from typing import List, Optional
+
+from theseus.geometry.functional import se3
+from .robot import Robot
+from .link import Link
+from .joint import Joint
+
+# TODO: Add support for joints with DOF>1
+
+
+def ForwardKinematicFactory(robot: Robot, link_names: Optional[List[str]] = None):
```

Comment:
The `link_names` are links that we want to get their poses through FK. In some cases, we don't want the poses for all the links. This will simplify the computation.